packages/controller/src/main.ts

```diff
@@ -5740,6 +5757,35 @@ async function startUpgradeManager(options: UpgradeArguments): Promise<void> {
     upgradeProcess.unref();
 }
 
+/**
+ * Checks if a system reboot is required and generates a notification if this is the case
+ */
+async function checkRebootRequired(): Promise<void> {
```

Comment:
Please add a return here if it is not a linux system, makes no sense to even try on darwin or win ... also freebsd no idea